src/esp/metadata/attributes/SceneAttributes.h

```diff
@@ -101,10 +103,93 @@ class SceneObjectInstanceAttributes : public AbstractAttributes {
   void setShaderType(int shader_type) { setInt("shader_type", shader_type); }
   int getShaderType() const { return getInt("shader_type"); }
 
+  /**
+   * @brief Get or set the uniform scaling of the instanced object.
+   */
+  float getUniformScale() const { return getFloat("uniform_scale"); }
+  void setUniformScale(float uniform_scale) {
+    setFloat("uniform_scale", uniform_scale);
+  }
+
+  /**
+   * @brief Get or set the mass scaling of the instanced object.
+   */
+  float getMassScale() const { return getFloat("mass_scale"); }
+  void setMassScale(float mass_scale) { setFloat("mass_scale", mass_scale); }
+
  public:
   ESP_SMART_POINTERS(SceneObjectInstanceAttributes)
 };  // class SceneObjectInstanceAttributes
 
+/**
+ * @brief This class describes an instance of an articulated object in a scene -
+ * along with its template name, translation from the origin, rotation,
+ * motiontype, and other values inherited from SceneObjectInstanceAttributes, it
+ * also holds initial joint pose and joint velocities.
+ */
+class SceneAOInstanceAttributes : public SceneObjectInstanceAttributes {
+ public:
+  /**
+   * @brief SceneObjectInstanceAttributes handle is also the handle of the
+   * underlying @ref AbstractObjectAttributes for the object being instanced.
+   */
+  explicit SceneAOInstanceAttributes(const std::string& handle);
+
+  /**
+   * @brief Articulated Object Instance only. Get or set whether or not base is
+   * fixed.
+   */
+  bool getFixedBase() const { return getBool("fixed_base"); }
+  void setFixedBase(bool fixed_base) { setBool("fixed_base", fixed_base); }
+
+  /**
+   * @brief retrieve a mutable reference to this scene attributes joint initial
+   * pose map
+   */
+  std::map<std::string, float>& getInitJointPose() { return initJointPose_; }
```

Comment:
const function? 
Check the other functions as well.

Reply:
returns a mutable reference